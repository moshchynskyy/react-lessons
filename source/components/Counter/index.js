// Core
import React from 'react';
import PropTypes from 'prop-types';

// Styles
import Styles from './styles.m.css';

const Counter = ({ count }) => (
    <section className = { Styles.counter } >Posts: { count }</section>
);

Counter.propTypes = {
    count: PropTypes.number.isRequired,
};

Counter.defaultProps = {
    count: 0,
};

export default Counter;