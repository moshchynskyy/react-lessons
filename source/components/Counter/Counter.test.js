// Core
import React from 'react';
import Counter from './';
import renderer from 'react-test-renderer';

const renderTree = renderer.create(<Counter count = { 3 } />).toJSON();

test('Counter should correspond to its snapshot counterpart 1...', () => {
    expect(renderTree).toMatchSnapshot();
});