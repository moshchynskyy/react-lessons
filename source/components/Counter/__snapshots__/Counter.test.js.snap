// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter must render ... 1`] = `
<section
  className="counter"
>
  Posts: 
  3
</section>
`;

exports[`Counter should correspond to its snapshot counterpart 1... 1`] = `
<section
  className="counter"
>
  Posts: 
  3
</section>
`;
